### PR TITLE
Add Guild#me

### PR DESCRIPTION
**Pull request by aemino**
(also did minor docs cleanup)

**Please describe the changes this PR makes and why it should be merged:**
A handy convenience method that allows you to get the logged in user as a member of a guild.

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
- [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/ClientDataResolver.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ class ClientDataResolver {
    * Data that resolves to give a GuildMember object. This can be:
    * * A GuildMember object
    * * A User object
-   * @typedef {Guild} GuildMemberResolvable
+   * @typedef {GuildMember} GuildMemberResolvable
```
Comment (PgBiel) [inline src/client/ClientDataResolver.js]: If the point is to resolve a GuildMember then why give a GuildMember if you already have it...

Comment (PgBiel) [inline src/client/ClientDataResolver.js]: other than that I approve the actual #me part :/ (unrelated changes, sigh)

**Comment by motiontheking** [comment]
Don't feel like docs cleanup should be part of this

**Comment by PgBiel** [comment]
^

File: src/client/ClientDataResolver.js (hunk 1 above)
Comment (aemino) [inline src/client/ClientDataResolver.js]: The point of the resolver (which I did not write) is to resolve users to GuildMembers. GuildMembers are compatible just for code simplification...

Comment (PgBiel) [inline src/client/ClientDataResolver.js]: @aemino then change the PR name to "Docs cleanup & Add guild#me"

**Comment by iCrawl** [comment]
Like Gawdl3y said, still not sure about it being a getter.

I know, like you said "we have a lot of getters that could be props", but that doesn't mean we need to introduce more, does it?
Changing old ones in v12 should be easy to comply.

**Comment by aemino** [comment]
Technically this should be a getter.  If someone keeps a reference to a guild instance and the client member leaves the guild, Guild#me would then return the now invalid GuildMember object.

**Comment by GusCaplan** [comment]
can you remove the changes not related to the pr thx

**Comment by GusCaplan** [comment]
I feel like this pr focuses too much on itself ("me") and not enough on others. Why is it so egotistical?

File: src/structures/Guild.js [hunk 2]
```diff
@@ -321,6 +321,15 @@ class Guild {
   }
 
   /**
+   * Gets the client user as a GuildMember of this guild.
+   * @type {GuildMember}
```
Comment (GusCaplan) [inline src/structures/Guild.js]: technically this could be null

File: src/structures/Guild.js [hunk 3]
```diff
@@ -321,6 +321,15 @@ class Guild {
   }
 
   /**
+   * The client user as a GuildMember of this guild
+   * @type {GuildMember}
```
Comment (GusCaplan) [inline src/structures/Guild.js]: `{?GuildMember}` and remove all the changes not related to this pr, put them in another one

Comment (iCrawl) [inline src/structures/Guild.js]: No.
